agent/command/s3_put.go

```diff
@@ -189,7 +189,6 @@ func (s3pc *s3put) validate() error {
 		catcher.NewWhen(s3pc.AwsSecret == "", "AWS secret cannot be blank")
 	}
 
-	catcher.NewWhen(s3pc.AwsSessionToken != "" && s3pc.Visibility == artifact.Signed, "cannot use temporary AWS credentials with signed link visibility")
```

Comment:
We previously didn't save the role arn when we upload a file, so we had no way of signing it. This PR adds signing, and s3.put does do some logic to check if previous ec2.assume_role commands permissions were passed in.